### PR TITLE
Extend readme

### PR DESCRIPTION
**Pull request by tjadevries**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,5 +7,68 @@
...
+@dataclass(frozen=True)
+class ExampleConfig(ConfigBase):
+    """Config for an example"""
+
+    section1: ExampleConfigSection = ExampleConfigSection()
+
+    @staticmethod
+    def get_app_basename() -> str:
+        """Return the string that describes the application base name"""
+        return "example"
```
Comment (JJdeVries) [inline README.md:52]: I think we should should explicitly mention that the `get_app_basename` determines the folder in the HOME directory the config file should be placed.

Comment (JJdeVries) [reply README.md:52]: Maybe that could even go in the docstring? But it should at least explicitly be mentioned in the readme i think.

File: README.md [hunk 2]
```diff
@@ -7,5 +7,68 @@
...
+field2 = 22
+```
+
+### Use config parameters in your code
+
+```python
+the_config = ExampleConfig.get()
+a_variable: str = the_config.section1.field1  # a_variable == "my own version of field1"
+another_variable: int = the_config.section1.field2  # another_variable == 22
+```
```
Comment (JJdeVries) [inline README.md:74]: what currently happens if the type does not match? Eventually we'll probably want to raise custom exceptions, but for now I'm not fully sure what happens?

**Review by JJdeVries — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (tjadevries) [reply README.md:52]: Done

File: README.md (hunk 2 above)
Comment (tjadevries) [reply README.md:89]: added a test and a note in the readme

**Review by JJdeVries — APPROVED** [review]

**Pull request merged**